src/Application/Actions/Donations/Confirm.php

```diff
@@ -92,9 +91,28 @@ protected function action(Request $request, Response $response, array $args): Re
             // Note that `on_behalf_of` is set up on create and is *not allowed* on update.
         ]);
 
-        $this->stripeClient->paymentIntents->confirm($paymentIntentId, [
-            'payment_method' => $pamentMethodId,
-        ]);
+        try {
+            $this->stripeClient->paymentIntents->confirm($paymentIntentId, [
+                'payment_method' => $pamentMethodId,
+            ]);
+        } catch (CardException $exception) {
+            $this->logger->info(sprintf(
+                'Stripe CardException on Confirm for donation %s (%s): %s',
+                $donation->getUuid(),
+                $paymentIntentId,
+                $exception->getMessage(),
+            ));
+
+            $this->entityManager->rollback();
+
+            return new JsonResponse([
+                'error' => [
+                    'message' => $exception->getMessage(),
+                    'code' => $exception->getStripeCode(),
+                    'decline_code' => $exception->getDeclineCode(),
```

Comment:
Depending whether the messages stripe generates are in the language we want to present, it might be worth matching on the decline code and presenting our own copy, at least for the ones we expect will be most common. Listed at https://stripe.com/docs/declines/codes?locale=en-GB

Reply:
@bdsl The ones we've seen this AM say _Your card was declined._ I think simply presenting them to the donor will roughly mirror what Stripe.js did before.